apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -1134,11 +1141,13 @@ handle_info({force_update_presence, LUser}, StateName,
     NewStateData =
     case StateData#state.pres_last of
         #xmlel{name = <<"presence">>} ->
-            PresenceEl = ejabberd_hooks:run_fold(
+            Stanza = mongoose_stanza:from_kv(packet, StateData#state.pres_last),
```

Comment:
Yes, it needs to be sorted out. Actually it shouldn't be called 'element' either, because normally 'element' denotes a stanza which initially entered the system, which is not the case here. All those places are going to be revisited in stage II. By the way, I can't find any calls to force_update_presence, do we really need it?